tests/datasource/test_datasource_serialization.py

```diff
@@ -0,0 +1,45 @@
+"""Test v3 API datasource serialization."""
+import pytest
+
+from great_expectations.data_context.types.base import (
+    DatasourceConfig,
+    datasourceConfigSchema,
+)
+
+
+@pytest.mark.parametrize(
+    "datasource_config,expected_serialized_datasource_config",
+    [
+        pytest.param(
```

Comment:
These are some example configurations, these test cases can be extended and this pattern used for other config objects to test serialization of important config cases.